DependencyInjection/Configuration.php

```diff
@@ -81,13 +81,25 @@ public function getConfigTreeBuilder()
                                         ->ifTrue(function ($v) { isset($v['route_basepath']) && isset($v['route_basepaths']); })
                                         ->thenInvalid('Found values for both "route_basepath" and "route_basepaths", use "route_basepaths" instead.')
                                     ->end()
+                                    ->beforeNormalization()
+                                        ->ifTrue(function ($v) { return isset($v['route_basepath']); })
+                                        ->then(function ($v) {
+                                            @trigger_error('The route_basepath setting is deprecated as of version 1.4 and will be removed in 2.0. Use route_basepaths instead.', E_USER_DEPRECATED);
+
+                                            return $v;
```

Comment:
`route_basepath` -> `route_basepaths` transformation is already handled by the fixXmlConfig feature.